### PR TITLE
[WIP][11.0][IMP]Add maintainer(s) to Odoo modules

### PR DESCRIPTION
**Pull request by StephanRozendaal**
Add maintainers to the Odoo modules, they are set-up in the same way as the 'odoo.authors' are.

### CODE REVIEW

File: github_connector_odoo/models/odoo_maintainer.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
```
Comment (RoelAdriaans) [inline github_connector_odoo/models/odoo_maintainer.py:2]: Is that your name? :)

**Comment by RoelAdriaans** [comment]
I think this should be in the github_connector_oca module, since the `maintainers` key is only in their module ( https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L12 )
and google docs document: https://docs.google.com/document/d/1ewkGkHzaKgtvnZZhp0ugSHiKZhlo3lechmGwsJ4OdNc/edit#

But Odoo has the `maintainer` key, but that looks like a lot of garbage.. :)

Which one should we use?
(CC @sbidoul @elicoidal )

**Review by RoelAdriaans — CHANGES_REQUESTED** [review]

File: github_connector_odoo/models/github_organization.py [hunk 2]
```diff
@@ -14,4 +14,6 @@ class GithubOrganization(models.Model):
 
     default_author_text = fields.Char(string='Default Author Text')
 
+    default_maintainer_text = fields.Char(string='Default Maintainer Text')
```
Comment (elicoidal) [inline github_connector_odoo/models/github_organization.py:17]: string is not necessary

File: github_connector_odoo/models/odoo_maintainer.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from odoo import api, fields, models
+
+
+class OdooMaintainer(models.Model):
+    _name = 'odoo.maintainer'
+    _order = 'module_qty desc, name'
+
+    # Column Section
+    name = fields.Char(
+        string='Name', store=True, readonly=True, index=True)
```
Comment (elicoidal) [inline github_connector_odoo/models/odoo_maintainer.py:14]: strings are not necessary if same name as column

File: github_connector_odoo/models/odoo_maintainer.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class OdooMaintainer(models.Model):
+    _name = 'odoo.maintainer'
```
Comment (elicoidal) [inline github_connector_odoo/models/odoo_maintainer.py:9]: add docstrings

**Review by elicoidal — CHANGES_REQUESTED** [review]
Main problem is all these modifications should be done in the OCA specific module

**Comment by pedrobaeza** [comment]
Please rebase and solve conflicts.

**Review by legalsylvain — COMMENTED** [review]
Hi @StephanRozendaal. 

Thanks a lot for this great addition  !

~About the position of this code I'm more in favour to keep it in this module because it's a good practice to add maintainers that will be followed not only by the OCA, but possibly by all the community. (I mean i think that big Odoo integrators will use this norm for the custom project / extra OCA project). I'll personaly will use it for module that are not under the OCA umbrella for the time being.~
About that point, I didn't saw the new oca module, in the same repository with maturity concept introduced. Well in that case, moving the code is indeed maybe the better choice.

About the design, I'm asking myself if it's not better to use the res.partner model instead of creating a new model "odoo.maintainer". The maintainer is a github login, and res.partner has github login since it inherits from abstract_github_model. 
What do you think ?
As partner are associated with companies (via partner_id) We could then answer to the question : how many modules are maintained by CampToCamp ? Acsone ? Akretion ? etc. 
It's a good corporate information.
What do you think ?

CC : @OCA/board.

> But Odoo has the maintainer key, but that looks like a lot of garbage.. :)

in a very few modules, indeed.

```
./odoo/addons/test_converter/__manifest__.py:    'maintainer': 'OpenERP SA',
./odoo/addons/test_assetsbundle/__manifest__.py:    'maintainer': 'Odoo SA',
./odoo/addons/test_pylint/__manifest__.py:    'maintainer': 'Odoo SA',
./addons/l10n_tr/__manifest__.py:    'maintainer':'https://launchpad.net/~openerp-turkey',
./addons/l10n_cn_small_business/__manifest__.py:    'maintainer': 'jeff@osbzr.com',
./addons/l10n_cn/__manifest__.py:    'maintainer': 'jeff@osbzr.com',
./addons/auth_crypt/__manifest__.py:    'maintainer': 'Odoo SA',
./addons/auth_oauth/__manifest__.py:    'maintainer': 'Odoo S.A.',
```
We can easily ignore it if maintainer contain a space, or @ or http, as maintainer should be a list of github login only.

**Comment by elicoidal** [comment]
> About the design, I'm asking myself if it's not better to use the res.partner model instead of creating a new model "odoo.maintainer". The maintainer is a github login, and res.partner has github login since it inherits from abstract_github_model.
What do you think ?

Agree with the idea indeed

**Comment by RoelAdriaans** [comment]
Connecting to res.partner could be an idea, but for CampToCamp, are modules maintained by the individual github users, or directly linked to the @camptocamp account?

If it's an individual, the res.partner can be linked  the the main partner, via the parent_id. 
What if that persons leaves camptocamp but keeps maintaining that module?

(And maybe for camptocamp it's not an issue because they have superate github accounts for their developers, but this is a hypothetical case..)

**Comment by elicoidal** [comment]
The users are linked to the company as followed:
screenshot from 2018-10-06 17-26-16

So I think it is fine.
If they leave the company,  you can have to cases:
* github account is company related ("seb-elico", "darshan-serpentcs")
* they are linked to the individual only

Depending on the case the attribution would follow either the company or the individual: nothing we can really do about it

File: github_connector_odoo/models/odoo_module_version.py [hunk 5]
```diff
@@ -103,16 +103,28 @@ class OdooModuleVersion(models.Model):
...
+    maintainer = fields.Char(string='Maintainer (Manifest)', readonly=True)
+
     author_ids = fields.Many2many(
         string='Authors', comodel_name='odoo.author',
         relation='github_module_version_author_rel',
         column1='module_version_id', column2='author_id', multi='author',
         compute='_compute_author', store=True)
 
+    maintainer_ids = fields.Many2many(
+        string='Maintainers', comodel_name='odoo.maintainer',
```
Comment (sbidoul) [inline github_connector_odoo/models/odoo_module_version.py:115]: I also think this should be a m2m to the res.partner with corresponding github login.

**Comment by MiquelRForgeFlow** [comment]
@StephanRozendaal what is the status of this PR? would you rebase it, resolve conflicts and attend last comments?

**Comment by pedrobaeza** [comment]
Is this still needed?

**Comment by OpenCode** [comment]
Is this still needed?

**Pull request closed without merging**